### PR TITLE
BC-2931 - Show task card in Task Overview page

### PR DESCRIPTION
**Pull request by MartinSchuhmacher**
# Short Description
Summary of making beta tasks (tasks with task card) appear correctly in task overview.

## Links to Ticket and related Pull-Requests
BC-2931

https://github.com/hpi-schul-cloud/schulcloud-client/pull/3147

https://github.com/hpi-schul-cloud/dof_app_deploy/pull/498

## Changes
- beta tasks lead to correct task card view from task overview
- beta tasks appear different then old tasks (in coordination with UX)
- beta tasks can be created via task overview
- task card page has different breadcrumbs if no course is chosen yet
- during creation mode coming from tasks overview the course can
be chosen before saving
- course selection is disabled once course is saved for task card

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/molecules/TaskItemStudent.unit.js [hunk 1]
```diff
@@ -44,13 +49,13 @@ describe("@/components/molecules/TaskItemStudent", () => {
 		});
 	};
 
-	it("Should link list item links to task/<id> page", () => {
+	it("Should link regularly to page for old tasks", () => {
```
Comment (odalys-dataport) [inline src/components/molecules/TaskItemStudent.unit.js:52]: maybe clearer wording like below would be nicer "should direct user to legacy task details page"

File: src/components/molecules/TaskItemStudent.vue [hunk 2]
```diff
@@ -4,48 +4,63 @@
 			:key="task.id"
 			v-click-outside="() => handleFocus(false)"
 			:href="href"
-			class="mx-n4 mx-sm-0"
```
Comment (odalys-dataport) [inline src/components/molecules/TaskItemStudent.vue:7]: you can have normal string classes and computed classes simultaneously, also we can use optional class assignment and then get rid of the getStyleClasses function

```
class="mx-n4 mx-sm-0"
:class="{ 'beta-task-background': isBetaTask }"
```

File: src/components/molecules/TaskItemStudent.vue [hunk 3]
```diff
@@ -4,48 +4,63 @@
...
-					@focus-changed="handleFocus"
-				/>
-			</v-list-item-action>
+			<template v-if="isBetaTask">
+				<v-list-item-avatar>
+					<v-icon class="fill" :color="iconColor">{{ taskIcon }}</v-icon>
+				</v-list-item-avatar>
+				<v-list-item-content>
+					<v-list-item-subtitle data-testid="taskSubtitle">{{
+						taskLabel
```
Comment (odalys-dataport) [inline src/components/molecules/TaskItemStudent.vue:20]: kinda weird formatting, maybe put curly brackets in the same line

File: src/components/molecules/TaskItemTeacher.unit.js [hunk 4]
```diff
@@ -63,7 +64,7 @@ describe("@/components/molecules/TaskItemTeacher", () => {
 		});
 	});
 
-	it("should compute correct href value", () => {
+	it("should compute regular href value for old tasks", () => {
```
Comment (odalys-dataport) [inline src/components/molecules/TaskItemTeacher.unit.js:67]: same as TaskItemStudent.unit.js

File: src/components/molecules/TaskItemTeacher.vue [hunk 5]
```diff
@@ -3,108 +3,125 @@
...
+						class="hidden-sm-and-up text--primary text-wrap"
 					>
+						<v-chip
+							v-if="hasUnpublishedLesson"
+							x-small
+							data-testid="task-lesson-chip-small"
+						>
+							{{
+								$t("components.molecules.TaskItemTeacher.lessonIsNotPublished")
+							}}</v-chip
```
Comment (odalys-dataport) [inline src/components/molecules/TaskItemTeacher.vue:60]: next line for v-chip closing tag

File: src/components/molecules/TaskItemTeacher.vue [hunk 6]
```diff
@@ -3,108 +3,125 @@
 		<v-list-item
 			:key="task.id"
 			v-click-outside="() => handleFocus(false)"
-			class="mx-n4 mx-sm-0"
+			:class="getStyleClasses"
```
Comment (odalys-dataport) [inline src/components/molecules/TaskItemTeacher.vue:6]: same comments as in TaskItemStudent.vue

File: src/components/molecules/TaskItemTeacher.vue [hunk 7]
```diff
@@ -3,108 +3,125 @@
...
+					v-else-if="showTaskStatus"
+					data-testid="task-status"
+					class="mr-8"
+				>
+					<v-list-item-action class="hidden-xs-only pl-4">
+						<v-list-item-subtitle>{{
+							$t("components.molecules.TaskItemTeacher.submitted")
+						}}</v-list-item-subtitle>
+						<v-list-item-title data-testid="taskSubmitted"
+							>{{ task.status.submitted }}/{{
```
Comment (odalys-dataport) [inline src/components/molecules/TaskItemTeacher.vue:93]: weird formatting again, but maybe it's just my opinion

File: src/components/templates/TasksDashboardMain.vue [hunk 8]
```diff
@@ -243,7 +244,36 @@ export default {
 			};
 		},
 		fabItems() {
+			const actions = [];
 			if (!this.isStudent) {
```
Comment (odalys-dataport) [inline src/components/templates/TasksDashboardMain.vue:248]: we should check for HOMEWORK_CREATE permission here as well

File: src/pages/TaskCard.page.vue [hunk 9]
```diff
@@ -13,7 +13,7 @@
 				item-value="id"
 				item-text="title"
 				filled
-				disabled
+				:disabled="isCreationMode ? false : !!course"
```
Comment (odalys-dataport) [inline src/pages/TaskCard.page.vue:16]: maybe we can merge this condition into isCreationMode and name it isCourseDisabled

**Review by odalys-dataport — CHANGES_REQUESTED** [review]

File: src/components/molecules/TaskItemStudent.vue (hunk 3 above)
Comment (MartinSchuhmacher) [reply src/components/molecules/TaskItemStudent.vue:20]: I tried putting it in the same line with different variants but then the label always gets several lines due to the line breaks, so I would prefer to leave it like this so the value stays correct

File: src/pages/TaskCard.page.vue (hunk 9 above)
Comment (MartinSchuhmacher) [reply src/pages/TaskCard.page.vue:16]: dont know what you mean by this comment. what exactly shoud I merge into `isCreationMode` ?

File: src/router/routes.ts [hunk 10]
```diff
@@ -253,6 +253,12 @@ export const routes: Array<RouteConfig> = [
 		component: () => import("../pages/TaskOverview.page.vue"),
 		name: "tasks",
 	},
+	{
+		path: `/tasks/create-task-card`,
```
Comment (christian-darsow) [inline src/router/routes.ts:257]: Lets change the path and name of the route to be in sync with the changes in BC-3613
path: /tasks/create-beta-task
name: tasks-beta-task-new

**Review by odalys-dataport — APPROVED** [review]

**Review by odalys-dataport — DISMISSED** [review]

**Review by christian-darsow — APPROVED** [review]

**Pull request merged**